tests/unit/test-cmd/test.run.js

```diff
@@ -326,101 +326,91 @@ describe('run', () => {
            sourceDir: '/fake/source/dir',
          },
        );
-     }
-  );
+     });
 
   describe('profile-create-new option', () => {
     beforeEach(() => {
       sinon.stub(fs, 'mkdir');
+      sinon.stub(fs, 'existsSync');
     });
 
     afterEach(() => {
       fs.mkdir.restore();
+      fs.existsSync.restore();
     });
 
-    it('creates dir when firefox profile doesn\'t exist',
-       async () => {
-         sinon.stub(fs, 'existsSync').returns(false);
-         const firefoxProfile = '/pretend/path/to/Firefox/profile';
-         const cmd = prepareRun();
-
-         await cmd.run({
-           firefoxProfile,
+    const fakeFirefoxProfile = '/pretend/path/to/Firefox/profile';
+    const fakeChromiumProfile = '/pretend/path/to/Chromium/profile';
+
+    async function testCreateProfileIfMissing(
+      existsSyncReturnValue,
+      runParams,
+      expectedProfile
+    ) {
+      fs.existsSync.returns(existsSyncReturnValue);
+      const cmd = prepareRun();
+
+      await cmd.run(runParams);
+
+      if (existsSyncReturnValue) {
+        sinon.assert.notCalled(fs.mkdir);
+      } else {
+        sinon.assert.calledWith(fs.mkdir, expectedProfile);
+      }
+
+      if (expectedProfile === fakeFirefoxProfile) {
```

Comment:
another option to decide which extension runner we should run the assertion on would be to look to `runParams.target` and based on that (if it is == to `chromium`) run the assertion on the chromiumRunnerStub.

and by doing that I think that we would not need to use a separate fake profile dir for the two of them and we could just use one for both.


how that sounds?

Reply:
Sounds perfect. 👍